src/styles/blocks/calendar__day.scss

```diff
@@ -0,0 +1,31 @@
+.calendar__day {
+  @include dayCell ($cellSize, $cellColor);
+  transition: all 0.5s ease;
+
+  &::before {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 100%;
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+  }
+
+  @for $index from 1 through $daysNumber {
+    &:nth-child(#{$index})::before {
+      content: "#{$index}";
+    }
+  }
+
+  @for $index from ($daysNumber + 1) through 31 {
+    &:nth-child(#{$index}) {
+      display: none;
+    }
+  }
```

Comment:
You have some problems with month length. It doesn't change month length in browser. 
I think here is a problem.

Also I think that you should move @for to `.calendar`

```suggestion
  @for $index from 28 through 31 {
  calendar--month-length--#{$index}
    &:nth-last-child(-n + (31 - #{$index})) {
      display: none;
    }
  }
```